core/client/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -40,20 +44,37 @@
 public final class AlluxioBlockStore {
   private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
-  private static AlluxioBlockStore sClient = null;
+  // Only one block store will be kept for each master.
+  private static final Map<InetSocketAddress, AlluxioBlockStore> CACHED_BLOCK_STORES =
```

Comment:
Looks like I've already made AlluxioBlockStore as a per-master Singleton in this PR. Is there anything else you want me to do here?